documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -111,35 +121,194 @@ for fuzzy matches.
 **Finally**, we apply the ranking and sorting parameters, and "highlight" the
 fields that were matched in the results.
 
-The new search controller should allow for using different filters for the first
-step and to not use the full-text search. We should also create a new serializer
-for collection search requests. It should include the common parameters for
-`list` requests, such as `page` and `page_size`, and the parameters for the
-exact matches: `tag`, `creator` and `source`.
+The search controller needs to be updated to extract the first 2 steps into a
+`build_search_query` method. A new `build_collection_query` should be added and
+used when the `collection` parameter is present. This method should create a
+filter query for the relevant field and value.
+
+The pagination and dead link cleanup should be the same for additional search
+queries as for the default search queries.
+
+#### Search request serializer updates
+
+_The initial version of this plan proposed to add new API endpoints and use path
+parameters for the values (e.g., `/image/tag/cat`) to make the URLs more
+readable, easier to share, will be easier to cache or perform cache invalidation
+required by [#1969](https://github.com/WordPress/openverse/issues/1969).
+However, since tags and creator names contain characters that are special for
+path segments (`/`, `?` and `&`) that cannot be properly encoded, this approach
+turned out to be not feasible._
```

Comment:
Do you know if we captured the problem in an issue that can be linked here? I want to understand where the problem is precisely because it sounds pretty strange to me that that was an issue for the `tags/` endpoint 🤔 

Do we have an example of a URL involving the tag collection returning erroneous items? I may be missing something else, perhaps related to the casing.

Reply:
We first noticed this issue in the `/creator` endpoint: #3791
Then, we realized that tags has the same problem. You can see it now in prod for tags with a question mark.

You cannot enter this URL into browser's address bar: `https://api.openverse.engineering/v1/images/tag/18th century?/`, `?` is replaced with `/?/`, and thus the tag you're searching for becomes the one without a question mark.

You can access this tag if you URI-encode this path parameter: `https://api.openverse.engineering/v1/images/tag/18th%20century%3F/`. But as Sara noted elsewhere that encoding path parameters (not the query parameters) is unusual and unexpected. 
It becomes more confusing when you add query parameters to the URL, e.g. `https://api.openverse.engineering/v1/images/tag/18th century?/?page=2`.

In Nuxt, it becomes really confusing if a URL contains special symbols in the path parameter, and also has a query parameter, such as `http://localhost:8443/image/tag/tag? cat & dog?page=2`. The query is split in different places depending on what browser you use: Firefox or Chrome.
Note that currently these views don't support filtering using query parameters, and don't add parameters for `page` or `peaks`, but if we decide to add this in the future - it will be very problematic.

You can see my numerous attempts to make the special characters work in the creator and tag values in #3793

Reply:
I see the reasoning now, makes sense the changes 👍